api/models/Members.js

```diff
@@ -36,6 +36,17 @@ var hash_password = key => (member, cb) => {
   })
 }
 
+var handle_membership_change = callback => (member, cb) =>
```

Comment:
OK, so I know this lets us keep `beforeUpdate` pointfree, but I don't think it's the right move.

As far as I can tell, there is nothing about what `handle_membership_type` does that means it's natural for it to take 2 callbacks. It just does some stuff asynchronously, and then allows another computation to takeover. There's now a coupling between this function and the places that it's used which I don't think we need.

Now, if we to redefine various bits of ours/Sails API, there might be a neater way of doing it. Equally, if we thrash around with Ramda we can twist the API. In the end though, I think we might be better off with the slight bit of messiness in `beforeUpdate`